cmd/clusterctl/client/config/reader_viper.go

```diff
@@ -61,8 +61,15 @@ func (v *viperReader) Init(path string) error {
 	viper.AllowEmptyEnv(true)
 	viper.AutomaticEnv()
 
-	if err := viper.ReadInConfig(); err != nil {
-		return err
+	err := viper.ReadInConfig()
+	if err != nil {
+		// ConfigFileNotFoundError is returned if it cannot find the default
```

Comment:
Have you considered to make `os.Stat(path)` inside the `if path != "" { }`?

Reply:
I think we can do that. I'll refactor that.

Reply:
So I added the `os.Stat` check inside the `if path != "" {}`. 
I also added another commit that specifically checks all the supported extensions of the default config file and skips reading in the config if that doesn't exist. I think that may be more readable/understandable than having the special handling of `ConfigFileNotFoundError`.
If the second commit looks better let me know and I can squash them.

Reply:
Because I had to make another change, I just went ahead and squashed the commits. Let me know if this looks better.

Reply:
LGTM